test/client-errors.js

```diff
@@ -549,7 +549,9 @@ test('socket fail while ending request body', (t) => {
 
     const _err = new Error('kaboom')
     client.on('connect', () => {
-      client.socket.destroy(_err)
+      process.nextTick(() => {
```

Comment:
why this nextTick?

Reply:
Because otherwise the socket will be destroyed before the queue has been resumed, which kind of invalidates the test.

Reply:
changed to resume before emit.